.github/workflows/generate-index.yaml

```
@@ -0,0 +1,30 @@
+name: Generate index.json
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  generate-index:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.8
+
+      - name: Install dependencies
+        run: pip install pyyaml
+
+      - name: Run generate_index.py
+        run: python utils/generate_index.py
+
+      - name: Commit and push index.json
+        uses: stefanzweifel/git-auto-commit-action@v5
+        with:
+          commit_message: "Update index.json [auto]"
+          file_pattern: index.json
```

Rather than auto-committing to main as a follow up to PRs, I'd prefer if this job ran on PRs (and pushes to main) and failed loudly if running `generate_index.py` resulted in any changes. Contributors who make changes to (or add) interfaces would run the script and commit the changes in their PR.
```suggestion
      - name: Ensure index.json is up to date
        run: |
          STATUS=$(git status --porcelain)
          echo "$STATUS"
          test -z "$STATUS"

```

---

so now if we make changes we have to `tox -e build-json-schemas update-json-index`? 
I don't see why this can't be done automatically in the first place, but if we really want it to be manual, can we consolidate it in a single tox env?